diff --git a/modules/ECMQmLoader.cpp.in b/modules/ECMQmLoader.cpp.in
index fe0da7d..5aa3f9e 100644
--- a/modules/ECMQmLoader.cpp.in
+++ b/modules/ECMQmLoader.cpp.in
@@ -77,7 +77,11 @@ namespace {
             return false;
         }
 #else
+# ifdef Q_OS_MACOS
+        const QString fullPath = QStandardPaths::locate(QStandardPaths::AppLocalDataLocation, subPath);
+# else
         const QString fullPath = QStandardPaths::locate(QStandardPaths::GenericDataLocation, subPath);
+# endif
         if (fullPath.isEmpty()) {
             return false;
         }
